universe/Planet.cpp

```diff
@@ -184,13 +184,15 @@ UniverseObjectType Planet::ObjectType() const
 { return OBJ_PLANET; }
 
 std::string Planet::Dump() const {
+    TemporaryPtr<System> cur_system = GetSystem(SystemID());
```

Comment:
If you have to get another object to get some info, it's probably not something that should be in a Dump function. The idea is to output this object's state, not any related information from other objects.